homeassistant/config_entries.py

```diff
@@ -2894,14 +2894,20 @@ def async_get_supported_subentry_types(
 
     @callback
     def _async_abort_entries_match(
-        self, match_dict: dict[str, Any] | None = None
+        self,
+        match_dict: dict[str, Any] | None = None,
     ) -> None:
         """Abort if current entries match all data.
 
         Requires `already_configured` in strings.json in user visible flows.
         """
         _async_abort_entries_match(
-            self._async_current_entries(include_ignore=False), match_dict
+            [
+                entry
+                for entry in self._async_current_entries(include_ignore=False)
+                if entry.entry_id != self.context.get("entry_id")
```

Comment:
Should we check the source of the flow is `SOURCE_REAUTH` or `SOURCE_RECONFIGURE`?
@bdraco @MartinHjelmare WDYT?

Reply:
We want to avoid creating another config entry with the same data or updating an existing config entry so that it conflicts with another config entry. We don't need to abort for any conflict from updates to an existing entry with the config entry itself.

As long as we don't change the meaning of an "entry_id" item in the config entry context, from meaning it's the entry_id of the config entry that the flow is updating, I don't think the config entry source matters.